modules/react/action-bar/lib/ActionBar.tsx

```diff
@@ -1,37 +1,39 @@
 import * as React from 'react';
-import {createComponent, styled, StyledType} from '@workday/canvas-kit-react/common';
-import {commonColors, colors, space} from '@workday/canvas-kit-react/tokens';
-import {HStack, HStackProps, StackSpacing} from '@workday/canvas-kit-react/layout';
+import {createContainer} from '@workday/canvas-kit-react/common';
+import {Menu} from '@workday/canvas-kit-react/menu';
+import {useActionBarModel} from './useActionBarModel';
+import {ActionBarList} from './ActionBarList';
+import {ActionBarOverflowButton} from './ActionBarOverflowButton';
+import {ActionBarItem} from './ActionBarItem';
 
-interface ActionBarProps extends Omit<HStackProps, 'spacing'> {
-  spacing?: StackSpacing;
+export interface ActionBarProps {
+  /**
+   * The contents of the ActionBar. Can be `ActionBar` children or any valid elements.
+   */
+  children: React.ReactNode;
 }
 
-const ResponsiveHStack = styled(HStack)<HStackProps & StyledType>(({theme}) => ({
-  [theme.canvas.breakpoints.down('s')]: {
-    padding: space.s,
-    '> *:not(div)': {
-      flex: 1,
-    },
-  },
-}));
-
-export const ActionBar = createComponent('div')({
+export const ActionBar = createContainer('div')({
   displayName: 'ActionBar',
-  Component: (elemProps: ActionBarProps, ref, Element) => (
-    <ResponsiveHStack
-      ref={ref}
-      as={Element}
-      spacing="s"
-      depth={1}
-      background={commonColors.background}
-      borderTop={`solid 1px ${colors.soap400}`}
-      padding={`${space.s} ${space.xl} `}
-      position="fixed"
-      bottom={0}
-      left={0}
-      right={0}
-      {...elemProps}
-    />
-  ),
+  modelHook: useActionBarModel,
+  subComponents: {
+    List: ActionBarList,
+    Item: ActionBarItem,
+    OverflowButton: ActionBarOverflowButton,
+    /**
+     * The overflow menu of the ActionBar component. If there isn't enough room to render all the action
+     * items, the extra actions that don't fit will be overflowed into this menu.
```

Comment:
I would add this to the `mdx` file as well. This tells the consumer how to use this. I think we should also mention what is the threshold of number of items before the the list is collapsed.

Reply:
In order for this to work they have to add the prop `overflowButton` and also use the menu is that correct?